cpp/src/dataframe/type_dispatcher.hpp

```diff
@@ -83,6 +86,9 @@
 /* ----------------------------------------------------------------------------*/
 namespace cudf{
 
+// This pragma disables a compiler warning that complains about the valid usage	
+// of calling a __host__ functor from this function which is __host__ __device__
```

Comment:
That would assume that all possible functors should be device callable, which is not the case. 

Imagine a functor that has a `cudaMalloc` call in it. That would not be a valid `__host__ __device__` function. But it would be a valid use of the `type_dispatcher`. Hence, why the pragma is used. 

An alternative, but less desirable solution would be to provide two type_dispatchers, one for host code, one for device code.